zerver/lib/export.py

```diff
@@ -2140,19 +2149,33 @@ def _get_exported_s3_record(
     return record
 
 
-def _save_s3_object_to_file(
-    key: "Object",
-    output_dir: str,
-    processing_uploads: bool,
+@dataclass
+class S3DownloadsProcessState:
+    output_dir: str
+    processing_uploads: bool
+    bucket: "Bucket"
+
+
+s3_downloads_context: ContextVar[S3DownloadsProcessState] = ContextVar("s3_downloads_context")
```

Comment:
We are not using the ContextVar for its thread-safety, here -- since we use processes, not threads, for parallelism.  All we need is a global box which is serializable by pickle's dependency analysis, which we can set and get out of in the other process.  An array could be made to work, for instance.  We want it primarily for things which take some work to set up and can't be pickled (Bucket) or are large and don't change (the list of user-ids with consent) which we don't want to pass on every call.

So this may be overkill.  However, were this the threaded form, this is what we'd want.

Reply:
Maybe we should move this note into a code comment?